src/services/MessagesService.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ *
+ */
+
+import { isLeft } from "fp-ts/lib/Either";
+import * as winston from "winston";
+import { ProblemJson } from "../types/api/ProblemJson";
+import { GetMessagesByUserOKResponse } from "../types/api_client/getMessagesByUserOKResponse";
+import { MessageResponseWithContent } from "../types/api_client/messageResponseWithContent";
+import { ServicePublic } from "../types/api_client/servicePublic";
+import {
+  Message,
+  Messages,
+  toAppMessageWithContent,
+  toAppMessageWithoutContent
+} from "../types/message";
+import { Service, toAppService } from "../types/service";
+import { User } from "../types/user";
+import { IApiClientFactoryInterface } from "./iApiClientFactory";
+
+export default class MessagesService {
+  constructor(private readonly apiClient: IApiClientFactoryInterface) {}
+
+  public async getMessagesByUser(user: User): Promise<Messages> {
+    try {
+      const messagesPayload = await this.apiClient
+        .getClient(user.fiscal_code)
+        .getMessagesByUser();
+
+      const maybeApiMessages = GetMessagesByUserOKResponse.decode(
+        messagesPayload
+      );
+      if (isLeft(maybeApiMessages)) {
+        const maybeProblemJson = ProblemJson.decode(messagesPayload);
+
+        if (isLeft(maybeProblemJson)) {
+          return Promise.reject(new Error("Unknown response."));
```

Comment:
As a general practice, I suggest to not call Promise.reject() and Promise.resolve() directly so you can avoid to wrap everything in a try / catch block. If you wrap the whole function body in a promise

```
public async getMessagesByUser(user: User): Promise<Messages> {
   return new Promise((resolve, reject) => { .
    ... reject(new Error()) ... 
     ... resolve({...}) 
   }
}
```
every arised exeption is automatically translated into a promise rejection
ref. http://2ality.com/2016/03/promise-rejections-vs-exceptions.html